### PR TITLE
Invoice: Improve expiry display

### LINKED ISSUE #1509: Display expiry time of invoices in minutes, hours or days if useful
### Describe the feature

In activity screen the expiry time of invoices are shown in seconds. It should be converted to minutes, hours or days if useful.

### PR DESCRIPTION
**Pull request by myxmaster**
# Description

Instead of displaying the initially defined time until expiration, the remaining time is now displayed in a human readable form. Also made the layout more consistent by putting labels on the left and values on the right.

grafik

grafik

Before:

grafik

grafik

This fixes #1509.

This pull request is categorized as a:

- [ ] New feature
- [ ] Bug fix
- [ ] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance 
- [x] Other

## Checklist
- [x] I’ve run `yarn run tsc` and made sure my code compiles correctly
- [x] I’ve run `yarn run lint` and made sure my code didn’t contain any problematic patterns
- [x] I’ve run `yarn run prettier` and made sure my code is formatted correctly
- [x] I’ve run `yarn run test` and made sure all of the tests pass

## Testing

If you modified or added a utility file, did you add new unit tests?

- [ ] No, I’m a fool
- [x] Yes
- [ ] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [x] Android
- [ ] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [x] LND (REST)
- [ ] LND (Lightning Node Connect)
- [x] Core Lightning (c-lightning-REST)
- [ ] Core Lightning (Spark)
- [ ] Eclair
- [x] LndHub

### Locales
- [x] I’ve added new locale text that requires translations
- [x] I’m aware that new translations should be made on the Zeus [Transfix page](https://app.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [x] Contributors will need to run `yarn` after this PR is merged in
- [x] 3rd party dependencies have been modified:
    * verify that `package.json` and `yarn.lock` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

**Review by shubhamkmr04 — COMMENTED** [review]
utACK

**Review by shubhamkmr04 — CHANGES_REQUESTED** [review]
Screenshot 2023-08-03 at 8 29 20 PM
We are getting expiration time as 0 seconds in LNDHub. I am using Alby instance in this one

**Comment by kaloudis** [comment]
> Screenshot 2023-08-03 at 8 29 20 PM We are getting expiration time as 0 seconds in LNDHub. I am using Alby instance in this one

Yeah, as suspected we need to test this against all the backends

**Comment by myxmaster** [comment]
@shubhamkmr04 Kind of fixing this blindly, please check if it's working now.

**Comment by kaloudis** [comment]
@myxmaster can you fix the conflicts here?

This PR will pave the way for this to work with LNDHub and all the other backends https://github.com/ZeusLN/zeus/pull/1652

**Comment by kaloudis** [comment]
Instead of saying `4 days, 2 hours` and it refers to the time to expiration and not the absolute time it is valid for, perhaps it should say `4 days, 2 hours from now`

**Comment by kaloudis** [comment]
@myxmaster what's the raw formatting of that timestamp above?

File: utils/Bolt11Utils.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
+// based on light-bolt11-decoder: Copyright (c) 2021 bitcoinjs contributors, fiatjaf
```
Comment (kaloudis) [inline utils/Bolt11Utils.ts:1]: what enhancements have been made here that make it faster than the original library?

**Comment by kaloudis** [comment]
Looking good, tested on LND and CLN so far.

Do we need to display expiry on paid invoices?

**Comment by kaloudis** [comment]
Screenshot 2023-12-22 at 11 21 44

Furthermore, on an expired request, maybe we remove the third line and replace the title with 'Expired request' or 'Expired invoice'?

**Comment by shubhamkmr04** [comment]
Tested for LND and LNDHub in both iOS and Android. Working fine.

**Review by kaloudis — APPROVED** [review]
tACK

**Pull request merged**